### ISSUE TITLE
BYTEPATH #0 - Introduction

### ISSUE THREAD

**Issue by SSYGEN** [SSYGEN/blog#30]
### [15. Final](https://github.com/SSYGEN/blog/issues/29)

<br>

## Introduction

This tutorial series will cover the creation of a complete game with [Lua](https://www.lua.org/) and [LÖVE](https://love2d.org/). It's aimed at programmers who have some experience but are just starting out with game development, or game developers who already have some experience with other languages or frameworks but want to figure out Lua or LÖVE better.

The game that will be created is a mix of [Bit Blaster XL](http://store.steampowered.com/app/433950/) and [Path of Exile's Passive Skill Tree](https://www.pathofexile.com/passive-skill-tree). You can see more about the game on its Steam page here:

<br>

<p align="center">

<a href="https://store.steampowered.com/app/760330/BYTEPATH/">
<img src="https://i.imgur.com/DjJbRxa.png">
</a>

</p>

<br>

The game is simple enough that it can be covered in a number of articles without extending for too long, but with enough content that a beginner would feel uncomfortable with the code and end up giving up before finishing. 

It's also at a level of complexity that most game development tutorials don't cover. Most of the problems beginners have when starting out with game development have to do with scope. The usual advice is to start small and work your way up, and while that might be a good idea, if the types of projects you're interested in cannot be made any smaller then there are very few resources out there that attempt to guide you through the problems that come up.

In my case, I've always been interested in making games with lots and lots of items/passives/skills and so when I was starting out it was really hard to figure out a good way to structure my code so that I wouldn't get lost. Hopefully these tutorials can help someone with that.

<br>

## Requirements

Before you start there are some programming knowledge requirements:

* The basics of programming, like variables, loops, conditionals, basic data structures and so on;

* The basics of OOP, like knowing what classes, instances, attributes and methods are;

* And the very basics of Lua, [this quick tutorial](http://tylerneylon.com/a/learn-lua/) should be good enough.

Essentially this is not for people who are just getting started with programming in general. Also, this tutorial series will have exercises. If you've ever been in the situation where you finish a tutorial and you don't know what to do next it's probably because it had no exercises, so if you don't want that to happen here then I recommend at least trying to do them.

<br>

---

If you liked these tutorials and want to support the writing of more like these in the future:

* ### [BYTEPATH on Steam](http://store.steampowered.com/app/760330/BYTEPATH/)
* ### [BYTEPATH-tutorial on itch.io](https://ssygen.itch.io/bytepath-tutorial)

Buying the tutorial on itch.io gives you access to the game's full source code, answers to exercises from articles 1 through 9, the code separated by articles (what the code should look like at the end of each article) and a Steam key to the game.

**Comment by athros** [comment]
I’d recommend linking learnxinyminutes.com (https://learnxinyminutes.com/docs/lua/) rather than http://tylerneylon.com/a/learn-lua/ because the comments render everywhere.

**Comment by SSYGEN** [comment]
@athros Thanks for pointing that out, I changed it.

**Comment by swapagarwal** [comment]
Maybe use GitBook for this? 😄

**Comment by ryukinix** [comment]
I liked the idea behind GitBook. Compile the whole stuff in a single document would be nice, though maybe we lost things like gifs that are useul sometimes to explain and demonstrate ideas, behaviors.

**Comment by SSYGEN** [comment]
@swapagarwal @ryukinix I've looked into it but if I'm going to bother to use a platform other than github issues for my blog I'll probably just build my own site and serve content that way, since I'll have more control over it.

**Comment by upgradeQ** [comment]
<details> 
  <summary>Answers</summary>

- [Game Loop ](https://raw.githack.com/a327ex/BYTEPATH/master/tutorial/answers/01%20-%20Game%20Loop%20Answers.html)
- [Libraries ](https://raw.githack.com/a327ex/BYTEPATH/master/tutorial/answers/02%20-%20Libraries%20Answers.html)
- [Rooms and Areas ](https://raw.githack.com/a327ex/BYTEPATH/master/tutorial/answers/03%20-%20Rooms%20and%20Areas%20Answers.html)
- [Exercises ](https://raw.githack.com/a327ex/BYTEPATH/master/tutorial/answers/04%20-%20Exercises%20Answers.html)
- [Game Basics ](https://raw.githack.com/a327ex/BYTEPATH/master/tutorial/answers/05%20-%20Game%20Basics%20Answers.html)
- [Player Basics ](https://raw.githack.com/a327ex/BYTEPATH/master/tutorial/answers/06%20-%20Player%20Basics%20Answers.html)
- [Player Stats and Attacks ](https://raw.githack.com/a327ex/BYTEPATH/master/tutorial/answers/07%20-%20Player%20Stats%20and%20Attacks%20Answers.html)
- [Enemies ](https://raw.githack.com/a327ex/BYTEPATH/master/tutorial/answers/08%20-%20Enemies%20Answers.html)
- [Director and Gameplay Loop ](https://raw.githack.com/a327ex/BYTEPATH/master/tutorial/answers/09%20-%20Director%20and%20Gameplay%20Loop%20Answers.html)
</details>

**Comment by encetroc** [comment]
I actually learn how to code with your tutorial, this really opened my eye on how to think to solve a problem or add new features. I mean sure I studied coding at school but I didn't even know why we use classes or why we should use classes. but after following this it is all clear now.

**Comment by encetroc** [comment]
Thank you <3

**Comment by yangruihan** [comment]
Thank you, I love it very much, can I get authorization to translate it into Chinese? :-)

**Comment by a327ex** [comment]
Yes, you can. The entire tutorial is MIT licensed so you can do whatever with it.
